javascript/ql/lib/semmle/javascript/frameworks/Restify.qll

```diff
@@ -157,13 +166,182 @@ module Restify {
     ServerDefinition server;
 
     RouteSetup() {
-      // server.get('/', fun)
-      // server.head('/', fun)
-      server.ref().getAMethodCall(any(Http::RequestMethodName m).toLowerCase()) = this
+      server
+          .ref()
+          .getAMethodCall([
+              "del", "get", "head", "opts", "post", "put", "patch", "param", "pre", "use"
+            ]) = this
     }
 
-    override DataFlow::SourceNode getARouteHandler() { result.flowsTo(this.getArgument(1)) }
+    override DataFlow::SourceNode getARouteHandler() {
+      exists(DataFlow::Node arg |
+        (
+          // server.get('/', fun)
+          // server.get('/', fun1, fun2)
+          // server.get('/', [fun1, fun2])
+          // server.param('name', fun)
+          this.getMethodName() = ["del", "get", "head", "opts", "post", "put", "patch", "param"] and
+          arg = this.getAnArgument() and
+          not arg = this.getArgument(0)
+          or
+          // server.use(fun)
+          // server.use(fun1, fun2)
+          // server.use([fun1, fun2])
+          this.getMethodName() = ["use", "pre"] and
+          arg = this.getAnArgument()
+        ) and
+        (
+          not arg.asExpr() instanceof ArrayExpr and
+          result.flowsTo(arg)
+          or
+          arg.asExpr() instanceof ArrayExpr and
+          result.flowsTo(DataFlow::exprNode(arg.asExpr().(ArrayExpr).getAnElement()))
```

Comment:
```suggestion
          not arg.getALocalSource() instanceof DataFlow::ArrayCreationNode and
          result.flowsTo(arg)
          or
          result = arg.getALocalSource().(DataFlow::ArrayCreationNode).getAnElement()
```

Reply:
Thanks, I had to change `result = arg.getALocalSource().(DataFlow::ArrayCreationNode).getAnElement()` to `result.flowsTo(arg.getALocalSource().(DataFlow::ArrayCreationNode).getAnElement())` to make it work, but that account for cases where the array is created in a different statement. Thanks a lot! Didnt know about `getLocalSource`